tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Tensor.java

```diff
@@ -261,23 +251,33 @@
     } else {
       nremaining = data.remaining();
     }
-    Tensor<?> t = allocateForBuffer(dtype, shape, nremaining);
+    Tensor<T> t = allocateForBuffer(dtype, shape, nremaining);
     t.buffer().put(data);
     return t;
   }
 
+  public static <T extends TType> Tensor<T> allocate(DataType<T> dtype, Shape shape) {
```

Comment:
Needs some javadoc. Looks like this makes an empty buffer, but what does that mean for types like String? Are they populated by the null reference or the empty string etc?

Reply:
I have to admit that the need to add a proper Javadoc to all endpoints affected by this PR is on my TODO list. I was waiting for some people to try it our first but then the SIG decided we should better merge it and fix it after. Are you still OK with this?

Reply:
Yeah, I'm fine with you merging it without the extra javadoc. We need to make sure it's there before there is a release with this code in though, so maybe file an issue for it?

Reply:
good idea, I'll create an issue right after merging